crates/flox-rust-sdk/src/models/environment/path_environment.rs

```diff
@@ -200,48 +200,38 @@ where
     /// - Create a result link as gc-root.
     /// - Create a lockfile if one doesn't already exist, updating it with
     ///   any new packages.
-    async fn build(
-        &mut self,
-        _nix: &NixCommandLine,
-        system: &System,
-    ) -> Result<(), EnvironmentError2> {
+    async fn build(&mut self, flox: &Flox) -> Result<(), EnvironmentError2> {
```

Comment:
question blocking: dropping `_nix` (and the earlier decision to not use `_nix`) means that the `nix` being used to build the environment doesn't have any of our configuration, e.g. `NIX_SSL_CERT_FILE`. How are we planning to ensure the `nix` we use here is properly configured? Or is that being passed somehow or do we know we don't need to pass config?

Reply:
I think that's something that should be handled on the builder side (`build-env.sh`). I'm not sure what all configuration is required, but I'd be happy to add it to the builder script.

Reply:
I'm not sure if some of it's cruft, but I think this is where we're setting stuff up:
https://github.com/flox/flox/blob/93785d22b78e59e7a5dcaa4cdf1e18956c608e50/crates/flox-rust-sdk/src/flox.rs#L474

Since that's something we want for the entirety of the CLI, I don't think it makes sense to move it to `build-env.sh`. Then if we want to invoke `nix` to do anything else we have to duplicate the config. It's possible we could at some point let `pkgdb` handle all the config but I'm not sure we're to that point yet

Reply:
I also don't think we need to fix setting all the config in this PR, but I'd rather not drop code for the prior method of setting config without a clear plan for what we'll do instead

Reply:
As far as I can tell, this is the only real usage of the Nix CLI left in `flox`. There's a handful of Nix CLI calls that are in the codebase, but I don't think they're actually codepaths we use today. We also aren't dropping any code for the prior method, it still exists.

It's hard to say what the plan for Nix config should look like going forward without knowing what the plan is for what the next builder will look like. If I remember correctly @aakropotkin suggested that the next builder may not call the Nix CLI at all, in which case the config could be something that we pass along in a JSON blob to whatever this builder is.

Setting all the flags that we set in `build-env.sh` requires a PR to `runix` because it doesn't support all of those flags for `nix build` yet (`--file`, `--print-out-path`, `--arg`, `--argstr`). Can we merge this PR, make the update to `runix`, and then come back here to replace `build-env.sh` entirely with a call to the `Build` struct until we figure out a more coherent plan?

Reply:
Calling `build.run()` takes `nix` as an argument, so you'll have to revert all the places you drop `nix` in this PR, which is what I meant by dropping code.

Resolving this thread as I wanted to raise/discuss the issue, but I don't think it needs to block this code from getting in.

Reply:
It's a good callout, thanks for reminding me. I totally forgot that we have a bunch of config for Nix calls.

> Calling build.run() takes nix as an argument

Yes, but you can get a `nix` via `flox.nix()`, so the interface doesn't need to change.